### ISSUE TITLE
UI issues when too many cell tags are entered

### ISSUE THREAD

**Issue by williamstein** [jupyter/notebook#2459]
To replicate:
1. Show the tags toolbar: View --> Cell Toolbar --> Tags
2. Enter a string with a few dozen spaces in it and click "Add Tag".
3. See the first few tags, then the rest are cut off.  Moreover, the x's to delete tags are mysteriously missing.

I noticed this when doing my from-scratch React.js based implementation of the Jupyter notebook and re-implementing this feature.  On the top is what things should look like (in my humble opinion) and on the bottom is what they do look like in Jupyter now:

<img width="1272" alt="screen shot 2017-04-29 at 11 46 04 am" src="https://cloud.githubusercontent.com/assets/1276278/25558002/97ae2cb4-2cd2-11e7-9db0-d8816d6080c0.png">

Another UI comments which you may disagree with: The ... to pop up a modal dialog to edit the same list of tags is completely pointless and redundant.  It provides no extra functionality, and is itself quite confusing (since you would assume that it does provide extra functionality).

**Comment by takluyver** [comment]
@vidartf do you have time to look into this?

**Comment by vidartf** [comment]
I see this locally as well. It seems the `cell-tag` class needs `white-space: nowrap;` in order to avoid the X's disappearing. I can PR this.

The cell toolbar has a fixed size of 29 pixels, which is why it does not wrap the tags on overflow (the control heights are inherited as well). This was the case from before tags PR, I simply went along with it without asking any questions. Since the tags cannot overflow, the "[...]" button was added (along with a fade out of the tags to help indicate this, which can be seen with the nowrap style). See the original PR (#2048) for this discussion. Currently the "[...]" is visible all the time, but it might be possible to engineer some JS/CSS to make it only appear on overflow, which might be a better UX clue to its purpose.
